### PR TITLE
Improves cinematic code

### PR DESCRIPTION
**Pull request by MrPerson**
## About The Pull Request

Makes the /obj/screen/cinematic that the cinematics use GC properly.
Adds a global signal for when a cinematic tries to play. Kills the global list of cinematics.
Makes cinematics no longer loop through the global mob list to make sure that each mob is in fact actually in the global mob list.

### CODE REVIEW

File: code/datums/cinematic.dm [hunk 1]
```diff
@@ -38,49 +36,48 @@ GLOBAL_LIST_EMPTY(cinematics)
...
 
 	//Pause OOC
 	var/ooc_toggled = FALSE
 	if(is_global && stop_ooc && GLOB.ooc_allowed)
 		ooc_toggled = TRUE
 		toggle_ooc(FALSE)
 
-
-	for(var/mob/M in GLOB.mob_list)
-		if(M in watchers)
```
Comment (MrPerson) [inline code/datums/cinematic.dm]: I want to point out how fucking stupid this code is. For global cinematics (ie literally all of them) watchers === GLOB.mob_list.

**Comment by cacogen** [comment]
i'm not saying the following is a good idea or in the scope of this pull but i wish cinematics were toggleable because the nar-sie one sucks and i want to kill people watching them

File: code/datums/cinematic.dm (hunk 1 above)
Comment (AnturK) [reply code/datums/cinematic.dm]: Yeah, i have no idea what was i thinking here.

**Comment by Tharcoonvagh** [comment]
The hand is great

File: code/datums/cinematic.dm [hunk 2]
```diff
@@ -38,49 +36,48 @@ GLOBAL_LIST_EMPTY(cinematics)
 	var/stop_ooc = TRUE //Turns off ooc when played globally.
 
 /datum/cinematic/New()
-	GLOB.cinematics += src
 	screen = new(src)
 
 /datum/cinematic/Destroy()
-	GLOB.cinematics -= src
+	for(var/CC in watching)
```
Comment (ninjanomnom) [inline code/datums/cinematic.dm]: Also null the `watching` var

File: code/datums/cinematic.dm [hunk 3]
```diff
@@ -38,49 +36,48 @@ GLOBAL_LIST_EMPTY(cinematics)
...
+		if(!CC)
+			continue
+		var/client/C = CC
+		C.screen -= screen
 	QDEL_NULL(screen)
-	for(var/mob/M in locked)
+	QDEL_NULL(special_callback)
+	for(var/MM in locked)
```
Comment (ninjanomnom) [inline code/datums/cinematic.dm]: Null `locked`

File: code/datums/cinematic.dm [hunk 4]
```diff
@@ -38,49 +36,48 @@ GLOBAL_LIST_EMPTY(cinematics)
...
-				locked += M
+	for(var/MM in watchers)
+		var/mob/M = MM
+		M.notransform = TRUE //Should this be done for non-global cinematics or even at all ?
+		locked += M
+		//Close watcher ui's
+		SStgui.close_user_uis(M)
+		if(M.client)
+			watching += M.client
+			M.client.screen += screen
```
Comment (ninjanomnom) [inline code/datums/cinematic.dm]: If there's no client you may want to listen for if a client joins mid cinematic

File: code/datums/cinematic.dm [hunk 5]
```diff
@@ -111,6 +108,11 @@ GLOBAL_LIST_EMPTY(cinematics)
 /datum/cinematic/proc/content()
 	sleep(50)
 
+/datum/cinematic/proc/replacement_cinematic(datum/source, datum/cinematic/other)
+	if(!is_global && other.is_global) //Allow it to play if we're local and it's global
+		return 0
```
Comment (ninjanomnom) [inline code/datums/cinematic.dm]:
```suggestion
		return NONE
```
or
```suggestion
		return
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/cinematic.dm (hunk 4 above)
Comment (MrPerson) [reply code/datums/cinematic.dm]: Good idea!

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**